javatests/arcs/core/data/BUILD

```diff
@@ -13,7 +13,7 @@ arcs_kt_jvm_test_suite(
     package = "arcs.core.data",
```

Comment:
You need to add the *Test.kt files as sources.

Reply:
arcs_kt_jvm_test_suite does that as a default.